app.py

```
@@ -439,13 +458,273 @@ def save_config():
+    if changed:
+        invalidate_dmx_mapping_cache()
+        save_config()
+
+
+_nl_hydrate_lock = threading.Lock()
+_nl_hydrate_in_flight = False
+
+
+def _schedule_nanoleaf_hydrate() -> None:
+    """Run layout/token hydration off the GET /api/lights path."""
+    global _nl_hydrate_in_flight
```

Ruff (0.16.1)

[warning] Using the global statement to update `_nl_hydrate_in_flight` is discouraged

(PLW0603)